guillotina/db/transaction_manager.py

```diff
@@ -138,7 +138,9 @@ def lock(self):
                         raise
             except Exception:
                 # failsafe terminate to make sure connection is cleaned
-                if txn._db_conn is not None:
+                if txn._db_conn is None:
+                    raise
+                if txn._db_conn._con is None:
                     raise
                 try:
                     await self._storage.terminate(txn._db_conn)
```

Comment:
This is to prevent this error:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/guillotina/db/transaction_manager.py", line 128, in _close_txn
    await self._storage.close(txn._db_conn)
  File "/usr/local/lib/python3.7/site-packages/guillotina/db/storages/pg.py", line 796, in close
    await shield(self.pool.release(con, timeout=1))
  File "/usr/local/lib/python3.7/site-packages/asyncpg/pool.py", line 645, in release
    return await asyncio.shield(ch.release(timeout), loop=self._loop)
  File "/usr/local/lib/python3.7/site-packages/asyncpg/pool.py", line 216, in release
    raise ex
  File "/usr/local/lib/python3.7/site-packages/asyncpg/pool.py", line 206, in release
    await self._con.reset(timeout=budget)
  File "/usr/local/lib/python3.7/site-packages/asyncpg/connection.py", line 1115, in reset
    await self.execute(reset_query, timeout=timeout)
  File "/usr/local/lib/python3.7/site-packages/asyncpg/connection.py", line 273, in execute
    return await self._protocol.query(query, timeout)
  File "asyncpg/protocol/protocol.pyx", line 301, in query
  File "asyncpg/protocol/protocol.pyx", line 659, in asyncpg.protocol.protocol.BaseProtocol._check_state
asyncpg.exceptions._base.InterfaceError: cannot perform operation: another operation is in progress
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/asyncpg/pool.py", line 214, in release
    self._con.terminate()
AttributeError: 'NoneType' object has no attribute 'terminate'
```

Reply:
:+1: